e2e/packages/test-data/generateLogs.ts

```diff
@@ -1,20 +1,38 @@
 import { execa } from "execa";
 import {
   ClientConfig,
+  GetContractReturnType,
+  Hex,
+  PublicClient,
+  RpcLog,
+  WalletClient,
   createPublicClient,
   createWalletClient,
   encodeEventTopics,
-  getContract,
   http,
   isHex,
   numberToHex,
+  getContract,
+  Transport,
+  Chain,
 } from "viem";
 import { mudFoundry } from "@latticexyz/common/chains";
 import { storeEventsAbi } from "@latticexyz/store";
-import { privateKeyToAccount } from "viem/accounts";
+import { Account, privateKeyToAccount } from "viem/accounts";
 import IWorldAbi from "../contracts/out/IWorld.sol/IWorld.abi.json";
 
-export async function generateLogs(numRecords: number, rpc: string) {
+type WorldAbi = typeof IWorldAbi;
+
+type WorldContract = GetContractReturnType<
+  WorldAbi,
+  PublicClient<Transport, Chain>,
+  WalletClient<Transport, Chain, Account>
+>;
+
+export async function generateLogs(
```

Comment:
Is this helper overkill? The idea is that you can pass a list of systems to execute and receive the logs back. 

Ideally we would have a typed list of calls instead of the "hook" that executes the systems. Something like:

```Typescript
const calls: [keyof WorldContract["write"], any>][] = [
  ["set", [[69]]],
  ["push", [420]],
];
```